accesstoken2/src/app.ts

```diff
@@ -72,17 +54,28 @@ export class AccessTokenLambda implements LambdaInterface {
                 statusCode: 200,
                 body: JSON.stringify(accessTokenResponse),
             };
-        } catch (err: any) { //Todo dont want any
+        } catch (err: unknown) { //Todo dont want any
```

Comment:
I think we're adding complexity with this catch statement, I think all we can simplify this

Reply:
I don't know but I have taken this onboard

```
 } catch (err: unknown) { //Todo dont want any
            logError(err as CustomErrors);
            return getError(err as CustomErrors);
        }
        function logError(err: CustomErrors) {
            logger.error({
                statusCode: err.statusCode ?? 500,
                message: err?.message,
                err: err
            });
        }
        function getError(err: CustomErrors): APIGatewayProxyResult {
            if (err == undefined || err.statusCode == undefined) return { statusCode: 500, body: JSON.stringify("Unknown Error!")}

            return {
                statusCode: err.statusCode ?? 500,
                body: JSON.stringify({
                    message: err?.statusCode >= 500 ? "Server Error": err.message,
                    code: err.code || null,
                    errorSummary: err.getErrorSummary(),
                }),
            };
        }
```
This is close to your original and seems to work

Reply:
This is better but again I dont think we need a specific function to force the logger to log the error, we're adding no additional functionality by creating new functions to logError and getError

Reply:
I removed the methods, I thought it helped with the err declaration but `any` and `unknown` are pretty much in the same level